crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +356,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
+
+                    // If we receive a ProofMPCOutput transaction, verify that it's valid & create a system transaction
+                    // to store its output on the blockchain, so it will be available for the initiating user.
+                    if let ConsensusTransactionKind::SignatureMPCOutput(
+                        value,
+                        session_id,
+                        sender_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mut signature_mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match signature_mpc_manager {
+                            Some(mpc_manager) => {
+                                let signature_mpc_manager = mpc_manager.lock().await;
```

Comment:
There's some confusing shadowing happening here. `signature_mpc_manager` is first an `Option` that is matched and
renamed to `mpc_manager` and then when the lock is acquired it is renamed `signature_mpc_manager`? Please improve naming
here.